numba/typeinfer.py

```diff
@@ -478,8 +481,22 @@ def resolve(self, typeinfer, typevars, fnty):
                 return
 
         # Resolve call type
-        sig = typeinfer.resolve_call(fnty, pos_args, kw_args)
-
+        try:
+            sig = typeinfer.resolve_call(fnty, pos_args, kw_args)
+        except ForceLiteralArg as e:
+            folded = e.fold_arguments(self.args, self.kws)
+            requested = set()
+            unsatisified = set()
+            for idx in e.requested_args:
+                maybe_arg = typeinfer.func_ir.get_definition(folded[idx])
+                if isinstance(maybe_arg, ir.Arg):
+                    requested.add(maybe_arg.index)
+                else:
+                    unsatisified.add(idx)
+            if unsatisified:
+                raise TypingError("Cannot request literal type.", loc=self.loc)
```

Comment:
note to self, `loc` use is checked and correct, exception chain contains the expected.